src/Microsoft.AspNetCore.Diagnostics/Internal/DiagnosticsLoggerExtensions.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Diagnostics.Internal
+{
+    internal static class DiagnosticsLoggerExtensions
+    {
+        private static readonly Action<ILogger, Exception, Exception> _unhandledException;
+        private static readonly Action<ILogger, Exception> _responseStartedErrorHandler;
+        private static readonly Action<ILogger, Exception, Exception> _errorHandlerException;
+        private static readonly Action<ILogger, Exception> _responseStartedErrorPageMiddleware;
+        private static readonly Action<ILogger, Exception, Exception> _displayErrorPageException;
+
+        static DiagnosticsLoggerExtensions()
+        {
+            _unhandledException = LoggerMessage.Define<Exception>(
+                LogLevel.Error,
+                1,
+                "An unhandled exception has occurred while executing the request: {Exception}");
+
+            _responseStartedErrorHandler = LoggerMessage.Define(
+                LogLevel.Warning,
+                2,
+                "The response has already started, the error handler will not be executed.");
+
+            _errorHandlerException = LoggerMessage.Define<Exception>(
+                LogLevel.Error,
+                3,
+                "An exception was thrown attempting to execute the error handler: {Exception}");
+
+            _responseStartedErrorPageMiddleware = LoggerMessage.Define(
+                LogLevel.Warning,
+                2,
```

Comment:
event id seems wrong

Reply:
It's from a different logger that uses the same logger message with event id 1 but different ones after it. I should add a comment to make that clear

Reply:
Here's what we have in SignalR for that case
https://github.com/aspnet/SignalR/blob/11cc57ee0e4ebc38c2cf5877b1df19e4e686f446/src/Microsoft.AspNetCore.SignalR.Client.Core/Internal/SignalRClientLoggerExtensions.cs#L82-L97